vcpkg & glib impasse

According to the vcpkg team, our glib 2.78.4 isn't compatible with the newer version of vcpkg's embedded python 3.12.

Our options are rolling back to an older vcpkg that uses python 3.11 or using the newer glib 2.80 (which is compatible with the latest vcpkg tools)

See https://github.com/microsoft/vcpkg/issues/41791#event-14908038274

I grepped for `2.78.4` to find where this glib version is defined, but I'm not getting any hits; how is this stuff defined?

Hunch is there's near zero risk with either approach, but I don't know how to progress it. Need help 📯

Now blowing up on libogg:

```
         Installing 20/27 libogg:x64-windows@1.3.5#1...
         Building libogg:x64-windows@1.3.5#1...
         C:\vcpkg\buildtrees\versioning_\versions\libogg\98e27727c32704393494d32615c25b7b16eb3067: info: installing overlay port from here
         -- Downloading https://gitlab.xiph.org/xiph/ogg/-/archive/v1.3.5/ogg-v1.3.5.tar.gz -> xiph-ogg-v1.3.5.tar.gz...
         [DEBUG] To include the environment variables in debug output, pass --debug-env
         [DEBUG] Trying to load bundleconfig from C:\vcpkg\vcpkg-bundle.json
         [DEBUG] Failed to open: C:\vcpkg\vcpkg-bundle.json
         [DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
         [DEBUG] Metrics enabled.
         [DEBUG] Feature flag 'binarycaching' unset
         [DEBUG] Feature flag 'compilertracking' unset
         [DEBUG] Feature flag 'registries' unset
         [DEBUG] Feature flag 'versions' unset
         [DEBUG] Feature flag 'dependencygraph' unset
     3>EXEC : warning : Download failed -- retrying after 1000ms [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : warning : Download failed -- retrying after 2000ms [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : warning : Download failed -- retrying after 4000ms [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : error : Missing xiph-ogg-v1.3.5.tar.gz and downloads are blocked by x-block-origin. [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : error : https://gitlab.xiph.org/xiph/ogg/-/archive/v1.3.5/ogg-v1.3.5.tar.gz: WinHttpSendRequest failed with exit code 12002 [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : error : https://gitlab.xiph.org/xiph/ogg/-/archive/v1.3.5/ogg-v1.3.5.tar.gz: WinHttpSendRequest failed with exit code 12002 [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : error : https://gitlab.xiph.org/xiph/ogg/-/archive/v1.3.5/ogg-v1.3.5.tar.gz: WinHttpSendRequest failed with exit code 12002 [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
     3>EXEC : error : https://gitlab.xiph.org/xiph/ogg/-/archive/v1.3.5/ogg-v1.3.5.tar.gz: WinHttpSendRequest failed with exit code 12002 [D:\a\dosbox-staging\dosbox-staging\src\libs\zmbv\zmbv.vcxproj]
         [DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030): 
         [DEBUG] Time in subprocesses: 0us
         [DEBUG] Time in parsing JSON: 6us
         [DEBUG] Time in JSON reader: 0us
         [DEBUG] Time in filesystem: 856us
         [DEBUG] Time in loading ports: 0us
         [DEBUG] Exiting after 1.5 min (91248420us)
         
         CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
```

This is a delicate clownshow :clown_face: !  Time to adjust the baseline a bit more (after the ogg fix but before the move to python 3.12)

--- a/.github/workflows/windows-msvc.yml
+++ b/.github/workflows/windows-msvc.yml
@@ -45,6 +45,16 @@ jobs:
         with:
           submodules: false
 
+      - name: Checkout vcpkg baseline
+        shell: pwsh
+        run: |
+          $baseline = (Get-Content vcpkg.json | ConvertFrom-Json).'builtin-baseline'
+          cd $env:VCPKG_INSTALLATION_ROOT
+          git fetch
+          rm vcpkg.exe
+          git -c advice.detachedHead=false checkout $baseline
+          bootstrap-vcpkg.bat -disableMetrics
+
       - name: Export GitHub Actions cache environment variables
         uses: actions/github-script@v7
         with:
@@ -111,6 +121,16 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: false
+
+      - name: Checkout vcpkg baseline
+        shell: pwsh
+        run: |
+          $baseline = (Get-Content vcpkg.json | ConvertFrom-Json).'builtin-baseline'
+          cd $env:VCPKG_INSTALLATION_ROOT
+          git fetch
+          rm vcpkg.exe
+          git -c advice.detachedHead=false checkout $baseline
+          bootstrap-vcpkg.bat -disableMetrics
 
       - name: Export GitHub Actions cache environment variables
         uses: actions/github-script@v7

--- a/vcpkg.json
+++ b/vcpkg.json
@@ -24,5 +24,5 @@
       "host": true
     }
   ],
-  "builtin-baseline": "0f6d4051514549541e61b186c4543341fd6ca8dd"
+  "builtin-baseline": "6ebf8c123bc8077dc9216db6438eeabaa154c473"
 }

